### PR TITLE
compiler: Compiler error when Link is set and Tooltip is an URL

### PR DESCRIPTION
**Pull request by alankan886**
fixes: https://github.com/terrastruct/d2/issues/888

### CODE REVIEW

**Comment by alankan886** [comment]
Some `.svg`, `.png` and `.pdf` files were "update" (although it looks the same) after I ran `./make.sh`, let me know if that's not expected. Also if the new tests I added are preferred in a different test file, lmk as well!

File: ci/release/changelogs/next.md
Comment (alixander) [inline ci/release/changelogs/next.md:7]: number is not right

File: d2compiler/compile.go
Comment (alixander) [inline d2compiler/compile.go:370]: "Tooltip cannot be set to URL when link is also set (for security)"

File: d2compiler/compile_test.go
Comment (alixander) [inline d2compiler/compile_test.go:1460]: good set of tests!

**Review by alixander — CHANGES_REQUESTED** [review]
@alankan886 thank you! 🚀 

The differing unrelated tests are due to computer architecture. I bet you're on x64 intel or some other AMD machine. These tests were generated on our ARM machines. Why is there a difference? TBD! https://github.com/terrastruct/d2/issues/921

In the meantime, could you uncheck these from Git? Sorry for the trouble.

Lastly, could you sign your commits please? https://github.com/terrastruct/d2/blob/master/docs/CONTRIBUTING.md#logistics

File: ci/release/changelogs/next.md
Comment (alixander) [reply ci/release/changelogs/next.md:7]: oh i see, you used the issue number and linked the PR. they should be the same number though, choose whichever one has more context

**Comment by alankan886** [comment]
Thanks for the quick review! I made the changes based on the comments, lmk if it's good!

**Review by alixander — CHANGES_REQUESTED** [review]
Screen Shot 2023-03-27 at 4 38 55 PM

@alankan886 can you squash your commits so that there's only the verified one? looks good other than that!

**Comment by alankan886** [comment]
One second, looks like the unwanted changes are back again.

**Review by alixander — APPROVED** [review]
thanks @alankan886 !

**Comment by alankan886** [comment]
@alixander oh nooooo I was about to squash again, I was struggling a bit because I was so used to using squash and merge on Github rip

**Comment by alixander** [comment]
hah appreciate that @alankan886 but no big deal =). it's a good point though i suppose i should wait for confirmation from the author after approving and before merging. will do next time.

and welcome to the community!

**Pull request merged**
